src/strictly_upper_triangular_logical_matrix.rs

```diff
@@ -49,6 +52,22 @@ pub struct StrictlyUpperTriangularLogicalMatrix {
     matrix: RoaringBitmap,
 }
 
+impl Eq for StrictlyUpperTriangularLogicalMatrix {}
+
+impl PartialEq for StrictlyUpperTriangularLogicalMatrix {
+    fn eq(&self, other: &Self) -> bool {
+        if self.size != other.size {
+            return false;
+        }
+        for (r, c) in RowColumnIterator::new(self.size) {
+            if self.get(r, c) != other.get(r, c) {
+                return false;
+            }
+        }
+        true
```

Comment:
This should avoid all the multiplications and additions done for indexing sake:

```suggestion
        self.iter_ones().eq(other.iter_ones())
```